src/include/dawg/root.h

```diff
@@ -15,17 +15,28 @@ namespace dawg {
 class root_model {
 public:
 	bool create(unsigned int len, const std::string &seq,  const std::vector<double> &rates) {
-		root_len = len;
+		root_len = (len == 0) ? seq.size() : len;
```

Comment:
Would something like this work?
```c++
if (seq.size() % 2 != 0) 
    len = seq.size();
else
    len = seq.size() / 3;
```

Reply:
If `seq.empty() == false` do we even need to access `root_len`?

Reply:
In `do_user_seq` the incoming sequence gets resized to the root length, so I think that's why that one example file you sent had no output.

Reply:
In what function do you want to add that code? `create` or `do_user_seq`?

Reply:
I was going to see if it could stay in `create`

Reply:
As written, it doesn't make sense in create since you are setting the value in `len` instead of reading from it.

`create` should just store `Root.Length` and `Root.Seq` and `do_user_seq` can rely on `residue_exchange` to convert the sequence.